### PR TITLE
Improve error message for failed tests

### PR DESCRIPTION
**Pull request by alexperovich**
https://github.com/dotnet/core-eng/issues/11558

### CODE REVIEW

File: tests/UnitTests.proj [hunk 1]
```diff
@@ -22,22 +22,22 @@
       <PayloadDirectory>$(MSBuildThisFileDirectory)Junit</PayloadDirectory>
     </HelixWorkItem>
   </ItemGroup>
-  <ItemGroup>
+  <!-- <ItemGroup>
```
Comment (MattGal) [inline tests/UnitTests.proj:25]: intentionally commented out?

Comment (alexperovich) [reply tests/UnitTests.proj:25]: Thats to test, will remove when I get confirmation that my error messages work.

**Review by ChadNedzlek — APPROVED** [review]
This will probably actually help a TON, since now the test failures get propagated to GitHub.  It would be really cool if we could get the line number of the test, so that the linking to source could work... is that in the results somewhere?

**Review by MattGal — APPROVED** [review]

**Comment by alexperovich** [comment]
The line number information might be. Though I don't think the xmls have it, so we don't report it.

**Pull request merged**